Modix.Services/UserInfo/UserInfoCommand.cs

```diff
@@ -0,0 +1,297 @@
+        public ulong UserId { get; }
+        public ulong GuildId { get; }
+        public IMessage Message { get; }
+    }
+
+    public class UserInfoCommandHandler : AsyncRequestHandler<UserInfoCommand>
+    {
+        private readonly ModixContext _modixContext;
+        private readonly IDiscordClient _discordClient;
+        private readonly IAutoRemoveMessageService _autoRemoveMessageService;
+        private readonly IImageService _imageService;
+        private readonly ILogger<UserInfoCommandHandler> _logger;
+        private readonly ModixConfig _modixConfig;
+        private readonly Stopwatch _stopwatch = new Stopwatch();
+
+        public UserInfoCommandHandler(ModixContext modixContext, IDiscordClient discordClient,
+            IAutoRemoveMessageService autoRemoveMessageService, IImageService imageService, ILogger<UserInfoCommandHandler> logger,
+            IOptions<ModixConfig> modixOptions)
+        {
+            _modixContext = modixContext;
+            _discordClient = discordClient;
+            _autoRemoveMessageService = autoRemoveMessageService;
+            _imageService = imageService;
+            _logger = logger;
+            _modixConfig = modixOptions.Value;
+        }
+
+        protected override async Task Handle(UserInfoCommand request, CancellationToken cancellationToken)
+        {
+            _stopwatch.Start();
+
+            _logger.LogDebug("Getting info for user {UserId} in guild {GuildId}", request.UserId, request.GuildId);
+
+            var guild = await _discordClient.GetGuildAsync(request.GuildId);
+
+            var discordUser = await guild.GetUserAsync(request.UserId);
+
+            _logger.LogDebug("Got user from Discord API");
+
+            var modixUser = await GetUserAsync(request.UserId, request.GuildId, cancellationToken);
+
+            if (discordUser is null
+                || modixUser is null)
+            {
+                // If we have nothing from Discord or from the database,
+                // we'll need to assume this user doesn't exist
+
+                _logger.LogDebug("Discord user or MODiX user is null, cannot return info, replying with retrieval error embed");
+
+                var embed = new EmbedBuilder()
+                    .WithTitle("Retrieval Error")
+                    .WithColor(Color.Red)
+                    .WithDescription("Sorry, we don't have any data for that user - and we couldn't find any, either.")
+                    .AddField("User Id", request.UserId);
+
+                await _autoRemoveMessageService.RegisterRemovableMessageAsync(
+                    request.Message.Author,
+                    embed,
+                    async (builder) =>
+                        await request.Message.ReplyAsync(string.Empty, embed: builder.Build()));
+
+                _stopwatch.Stop();
+
+                return;
+            }
+
+            var participation = await GetParticipationStatisticsAsync(guild, request.UserId, cancellationToken);
```

Comment:
This call (or at least awaiting the task that it returns) can be moved closer to where it's needed.